### ISSUE TITLE
Wrong time zone determined in calendar app

### ISSUE THREAD

**Issue by nerdoc** [nextcloud/server#1165]
### Steps to reproduce
1. Log in into calendar directly, by using the link https://<your_server_ip>/index.php/apps/calendar
2. watch the message that appears after login (in German): 
*"Du bist in einer unbekannten Zeitzone (Europe/Berlin), auf UTC wird geändert"*
3. see that any cal entry is wrong by 2 hours (here I am in Europe/Berlin) - I see an entry 17:00 - but I saved it as 19:00
4. Change to Files (Core) App
5. Change back to Calendar App
6. Now everything is in the correct time zone, it is recognized correctly. But only after switching the apps. The direct link to the calendar fails to recognize the correct TZ.

### Expected behaviour
Time zone should be correct.

### Actual behaviour
Time zone is not recognized.

### Server configuration
**Operating system**: CentOS 6 (uberspace.de)

**Web server:** Apache 2.4

**Database:** MySQL 5.5

**PHP version:** 7

**Nextcloud version:** 9.0.52 (stable)

**Updated from an older Nextcloud/ownCloud or fresh install:** Updated from Owncloud 9, the problem has been there as well, since v8
**Signing status:**
<details>
<summary>Signing status</summary>

```
No errors have been found.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
  - admin_audit: 1.0.0
  - calendar: 1.2.2
  - comments: 0.2
  - contacts: 1.3.1.0
  - dav: 0.1.6
  - documents: 0.12.0
  - encryption: 1.2.0
  - federatedfilesharing: 0.1.0
  - federation: 0.0.4
  - files: 1.4.4
  - files_external: 0.5.2
  - files_pdfviewer: 0.8.1
  - files_sharing: 0.9.1
  - files_texteditor: 2.1
  - files_trashbin: 0.8.0
  - files_versions: 1.2.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - gallery: 14.5.0
  - notifications: 0.2.3
  - ownnote: 1.08
  - password_policy: 1.0.0
  - provisioning_api: 0.4.1
  - systemtags: 0.2
  - tasks: 0.9.1
  - templateeditor: 0.1
  - theming: 0.1.0
  - updatenotification: 0.1.0
Disabled:
  - activity
  - external
  - user_external
  - user_ldap
  - user_saml
```
</details>

**The content of config/config.php:**
<details>
<summary>Config report</summary>

```
{
    "system": {
        "instanceid": "5261821e2dc17",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "datadirectory": "\/home\/nerdocs\/owncloud_data",
        "dbtype": "mysql",
        "version": "9.0.52.0",
        "dbname": "nerdocs_oc",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "theme": "",
        "maintenance": false,
        "default_language": "de",
        "logtimezone": "Europe\/Berlin",
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***"
        ],
        "secret": "***REMOVED SENSITIVE VALUE***",
        "loglevel": 3,
        "trashbin_retention_obligation": "auto",
        "mail_from_address": "owncloud",
        "mail_smtpmode": "php",
        "mail_domain": "***REMOVED SENSITIVE VALUE***"
    }
}
```
</details>

**Are you using external storage, if yes which one:** NO

**Are you using encryption:** yes

**Are you using an external user-backend, if yes which one:** NO

### Client configuration
**Browser:** Firefox 48, Chromium

**Operating system:** Ubuntu Linux

#### Web server error log
<details>
<summary>Web server error log</summary>
```
Nothing about nextcloud.
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>
```
{"reqId":"9HqA8LQJdC42T9TbR8EI","remoteAddr":"91.130.25.8","app":"no app in context","message":"Current user is not logged in","level":0,"time":"2016-08-29T17:11:06+02:00","method":"GET","url":"\/index.php\/apps\/calendar","user":"--"}
{"reqId":"MzUuWCIIZV\/vn+v\/BK2J","remoteAddr":"91.130.25.8","app":"admin_audit","message":"Login attempt: \"christian\"","level":1,"time":"2016-08-29T17:11:16+02:00","method":"POST","url":"\/index.php?redirect_url=%2Findex.php%2Fapps%2Fcalendar","user":"--"}
{"reqId":"MzUuWCIIZV\/vn+v\/BK2J","remoteAddr":"91.130.25.8","app":"admin_audit","message":"Login successful: \"christian\"","level":1,"time":"2016-08-29T17:11:16+02:00","method":"POST","url":"\/index.php?redirect_url=%2Findex.php%2Fapps%2Fcalendar","user":"christian"}
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>
```
a) The javascript console log:

Using //@ to indicate sourceMappingURL pragmas is deprecated. Use //# instead jquery.min.js:1
Content Security Policy: Die Einstellungen der Seite haben das Laden einer Ressource auf self blockiert ("script-src https://cloud.nerdocs.at 'unsafe-eval'").  calendar
getPreventDefault() sollte nicht mehr verwendet werden. Verwenden Sie stattdessen defaultPrevented. jquery.min.js:5:17117
```
<summary>User Agent</summary>
*Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:48.0) Gecko/20100101 Firefox/48.0*
</details>

Can I do anything else to help fix this bug?

**Comment by georgehrke** [comment]
There was an bug in the calendar app, that timezone detection failed on `index.php/apps/calendar` but worked on `index.php/apps/calendar/`.
That was fixed with calendar 1.3.0 or so.

**Comment by nerdoc** [comment]
Really. Updated to 1.4.0 and the problem is gone. Thanks.
